sql/logictest: TestLogic failed

[(sql/logictest).TestLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2421471&tab=buildLog) on [master@8489da91b75abcc63937550a576dbcdd215cb996](https://github.com/cockroachdb/cockroach/commits/8489da91b75abcc63937550a576dbcdd215cb996):

```
=== RUN   TestLogic
--- FAIL: TestLogic (927.85s)
    test_log_scope.go:154: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic996800531
    test_log_scope.go:63: use -show-logs to present logs inline
=== RUN   TestLogic/fakedist-spec-planning/aggregate/covariance
--- done: testdata/logic_test/aggregate with config fakedist-spec-planning: 254 tests, 2 failures
            --- FAIL: TestLogic/fakedist-spec-planning/aggregate/covariance (0.03s)
                logic.go:2150: 
                     
                    testdata/logic_test/aggregate:1475: SELECT covar_pop(y, x), covar_pop(int_y, int_x), covar_pop(y, int_x), covar_pop(int_y, x), covar_pop(dy, dx)::DECIMAL(10, 7),
                           covar_samp(y, x), covar_samp(int_y, int_x), covar_samp(y, int_x), covar_samp(int_y, x), covar_samp(dy, dx)::DECIMAL(9, 6)
                    FROM statistics_agg_test
                    expected:
                        -149.7003372 33 1100.4 -25.336322 -149.7003372 -166.333708 36.6666666666667 1222.66666666667 -28.1514688888889 -166.333708
                        
                    but found (query options: "") :
                        -149.700337200001  33  1100.4  -25.336322  -149.7003372  -166.333708000001  36.6666666666667  1222.66666666667  -28.1514688888889  -166.333708
                logic.go:1708: 
                     testdata/logic_test/aggregate:1482: too many errors encountered, skipping the rest of the input
=== RUN   TestLogic/fakedist-spec-planning/aggregate
=== PAUSE TestLogic/fakedist-spec-planning/aggregate
=== CONT  TestLogic/fakedist-spec-planning/aggregate
        --- FAIL: TestLogic/fakedist-spec-planning/aggregate (14.52s)
=== RUN   TestLogic/fakedist-spec-planning
    --- FAIL: TestLogic/fakedist-spec-planning (0.58s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #56269 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.2)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -2769,7 +2769,10 @@ func floatsMatch(expectedString, actualString string) (bool, error) {
 	if expPower != actPower {
 		return false, nil
 	}
-	for i := 0; i < 15; i++ {
+	// TODO(yuzefovich): investigate why we can't always guarantee deterministic
+	// 15 significant digits and switch back from 14 to 15 digits comparison
+	// here. See #56446 for more details.
+	for i := 0; i < 14; i++ {
 		expDigit := int(expected)
 		actDigit := int(actual)
 		if expDigit != actDigit {

